src/utils/encryption.ts

```diff
@@ -21,28 +28,28 @@ export class AES {
    * @param key - key to encrypt with
    * @returns ciphertext bundle
    */
-  static encryptGCM(plaintext: BytesData[], key: BytesData): Ciphertext {
+  static encryptGCM(plaintext: string[], key: string | Uint8Array): Ciphertext {
```

Comment:
BytesData includes `BN` and there are zero use cases where we pass `BN`, so it makes sense to demand more specific types, which in turn leads to less if-else's checking what kind of input we have.